pulpcore/tests/unit/serializers/test_repository.py

```diff
@@ -37,8 +40,45 @@ def test_validate_proxy_creds_update_invalid(self):
             serializer.is_valid(raise_exception=True)
         self.assertIn("can only be specified together", str(ctx.exception))
 
+    def _gen_remote_serializer(self):
+        data = {
+            **self.minimal_data,
+            "client_key": None,
+            "username": "user",
+            "password": "pass",
+            "proxy_url": "foobar",
+            "proxy_username": "proxyuser",
+            "proxy_password": "proxypass",
+        }
+
+        remote = FileRemote.objects.create(**data)
+        serializer = RemoteSerializer(remote, data=data, context={"request": None})
```

Comment:
```suggestion
        remote = FileRemote(**data)
        serializer = RemoteSerializer(remote, data=data, context={"request": None})
```

I think, this is what is trying to talk to the database. And for testing the serializer, there should be no need for. So with this suggested change, i believe we can all safely go back to just using unittest.

Reply:
Is the "SimpleNamespace" approach not working here?

Reply:
no, afaiu `hidden_fields` doesn't get serialized, because it's missing attributes
```
self = DetailIdentityField(view_name_pattern='remotes(-.*/.*)-detail')
obj = namespace(name='test', proxy_password='pass', proxy_url='http://whatever', proxy_username='user', url='http://whatever')

    def _view_name(self, obj):
        # this is probably memoizeable based on the model class if we want to get cachey
        try:
>           obj = obj.cast()
E           AttributeError: 'types.SimpleNamespace' object has no attribute 'cast'
```

Reply:
Ok, there's the second attempt to get to the database then. Can we mock `_view_name` here?

I kind of don't want this "easy" test to trigger the whole django create a test database machinery. And at the same time i don't want to invest too much into unittest style unit tests. (We have an ongoing test rewriting effort from unittest to pytest.)